cpp/include/cudf/strings/attributes.hpp

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright (c) 2019, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#pragma once
+
+#include <cudf/strings/strings_column_view.hpp>
+#include <cudf/column/column.hpp>
+
+namespace cudf
+{
+namespace strings
+{
+
+/**
+ * @brief Returns an integer numeric column containing the length of each string in
+ * characters.
+ *
+ * The output column will have the same number of rows as the
+ * specified strings column. Each row value will be the number of
+ * characters in the corresponding string.
+ *
+ * Any null string will result in a null entry for that row in the output column.
+ *
+ * @param strings Strings instance for this operation.
+ * @param mr Resource for allocating device memory.
+ * @return New INT32 column with lengths for each string.
+ */
+std::unique_ptr<cudf::column> characters_counts( strings_column_view strings,
```

Comment:
I think the item counted should be singular. There are multiple counts returned, which is already indicated by the plural counts.

```suggestion
std::unique_ptr<cudf::column> character_counts( strings_column_view strings,
```